vm/src/builtins/int.rs

```diff
@@ -157,18 +157,23 @@ fn inner_divmod(int1: &BigInt, int2: &BigInt, vm: &VirtualMachine) -> PyResult {
     Ok(vm.new_tuple((div, modulo)).into())
 }
 
-fn inner_shift<F>(int1: &BigInt, int2: &BigInt, shift_op: F, vm: &VirtualMachine) -> PyResult
+fn inner_shift<F, S>(
+    int1: &BigInt,
+    int2: &BigInt,
+    shift_op: F,
+    overflow_shift: S,
+    vm: &VirtualMachine,
+) -> PyResult
 where
     F: Fn(&BigInt, usize) -> BigInt,
+    S: Fn(&BigInt) -> usize,
```

Comment:
```suggestion
    S: Fn(&BigInt) -> PyResult<usize>,
```
The error has to be propagated out